tox.ini

```diff
@@ -79,7 +80,14 @@ deps = -r{toxinidir}/doc/requirements.txt
 commands = sphinx-build -a -E -W -d releasenotes/build/doctrees -b html releasenotes/source releasenotes/build/html
 
 [testenv:pylint]
-commands = pylint --rcfile=pylintrc bandit
+commands = -pylint --rcfile=pylintrc bandit
+
+[testenv:format]
+skip_install = true
+deps =
+    pre-commit
```

Comment:
pre-commit is great. It runs things like `pyupgrade` the way pyupgrade wants to be run (passing in each file independently). It can be used in local development as well and could run the rest of our linters for us for local development too